static/app/components/modals/recoveryOptionsModal.tsx

```diff
@@ -1,118 +1,116 @@
-import {Fragment} from 'react';
+import {Fragment, useState} from 'react';
 
 import type {ModalRenderProps} from 'sentry/actionCreators/modal';
 import {Alert} from 'sentry/components/alert';
 import {Button} from 'sentry/components/button';
-import DeprecatedAsyncComponent from 'sentry/components/deprecatedAsyncComponent';
 import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
 import type {Authenticator} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
 import TextBlock from 'sentry/views/settings/components/text/textBlock';
 
-type Props = DeprecatedAsyncComponent['props'] &
-  ModalRenderProps & {
-    authenticatorName: string;
-  };
-
-type State = DeprecatedAsyncComponent['state'] & {
-  authenticators: Authenticator[] | null;
-  skipSms: boolean;
+type Props = ModalRenderProps & {
+  authenticatorName: string;
 };
 
-class RecoveryOptionsModal extends DeprecatedAsyncComponent<Props, State> {
-  getDefaultState() {
-    return {...super.getDefaultState(), skipSms: false};
-  }
+function RecoveryOptionsModal({
+  authenticatorName,
+  closeModal,
+  Body,
+  Header,
+  Footer,
+}: Props) {
+  const {isLoading, data: authenticators = []} = useApiQuery<Authenticator[]>(
+    ['/users/me/authenticators/'],
+    {
+      staleTime: 5000, // expire after 5 seconds
+    }
+  );
+  const [skipSms, setSkipSms] = useState<boolean>(false);
 
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    return [['authenticators', '/users/me/authenticators/']];
-  }
-
-  handleSkipSms = () => {
-    this.setState({skipSms: true});
-  };
+  const {recovery, sms} = authenticators.reduce<{[key: string]: Authenticator}>(
+    (obj, item) => {
+      obj[item.id] = item;
+      return obj;
+    },
+    {}
+  );
 
-  renderBody() {
-    const {authenticatorName, closeModal, Body, Header, Footer} = this.props;
-    const {authenticators, skipSms} = this.state;
+  const recoveryEnrolled = recovery?.isEnrolled;
+  const displaySmsPrompt =
+    sms && !sms.isEnrolled && !skipSms && !sms.disallowNewEnrollment;
 
-    const {recovery, sms} = authenticators!.reduce<{[key: string]: Authenticator}>(
-      (obj, item) => {
-        obj[item.id] = item;
-        return obj;
-      },
-      {}
-    );
-    const recoveryEnrolled = recovery?.isEnrolled;
-    const displaySmsPrompt =
-      sms && !sms.isEnrolled && !skipSms && !sms.disallowNewEnrollment;
+  const handleSkipSms = () => {
+    setSkipSms(true);
+  };
 
-    return (
-      <Fragment>
-        <Header closeButton>{t('Two-Factor Authentication Enabled')}</Header>
+  if (isLoading) return null;
```

Comment:
Note: I did this cause it helps simplify the tests a little with the `findBy` -- is there a `Loading` component or anything I should use instead?

Reply:
we generally use `<LoadingIndicator />` for loading of bigger pages and `<Placeholder />` for smaller components within the big page